### ISSUE TITLE
Adjust tests so PR testing doesn't fail

### ISSUE THREAD

**Issue by lordjabez** [python-cachier/cachier#109]
Right now PR tests fail due to being unable to connect to Mongo. It would be nice if those tests could be ignored when not running against the main branch so the rest of CI could still indicate other problems if they exist.

Related to this, even running `pytest -m "not mongo"` still requires a functional Mongo instance because of the way the test code still tries to validate the Mongo connection outside of discrete Mongo-related tests. This is less than idea, even given how easy it is to run a MongoDB server in Atlas :) 

In summary:
1. Don't do any Mongo-related testing when `not mongo` is selected in `pytest`
2. Skip Mongo tests by default in non-`master` branch CI runs.
